### PR TITLE
Implementation of JAIL and DYNAMITE cards

### PR DESCRIPTION
**Pull request by bashnia007**

### CODE REVIEW

File: Bang/Players/Player.cs [hunk 1]
```diff
@@ -137,13 +143,15 @@ public void RegainLifePoint()
...
-        public void LoseLifePoint()
+        public void LoseLifePoint(int loseLifeAmount = 1)
         {
-            Debug.Assert(PlayerTablet.Health > 0);
-            
-            PlayerTablet.Health--;
-        }
+            for (int i = 0; i < loseLifeAmount; i++)
+            {
+                Debug.Assert(PlayerTablet.Health > 0);
```
Comment (IvanovAndrew) [inline Bang/Players/Player.cs:150]: Scenario: dynamite explodes on player with one health point. The debug assertion will be failed in the second iteration.

So I suppose to put debug assertion into method start.

File: Bang/GameEvents/CardEffects/DynamiteActionHandler.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using Bang.Players;
+using Bang.PlayingCards;
+
+namespace Bang.GameEvents.CardEffects
+{
+    internal class DynamiteActionHandler : CardActionHandler
+    {
+        public override HandlerState ApplyEffect(Game.Gameplay gameplay, Player victim, BangGameCard card)
+        {
+            Logger.Info($"Player {gameplay.PlayerTurn.Character.Name} player dynamite card");
```
Comment (IvanovAndrew) [inline Bang/GameEvents/CardEffects/DynamiteActionHandler.cs:11]: Small typo:  _player dynamite card_. Should be _played_ or _plays_

File: Bang/Game/Gameplay.cs [hunk 3]
```diff
@@ -150,5 +150,61 @@ public void ChooseCard(BangGameCard card, Player player)
         {
             state = state.ApplyReplyAction(player, card);
         }
+
+        public void StartNextPlayerTurn()
+        {
+            SetNextPlayer();
+
+            var dynamiteCard = PlayerTurn.PlayerTablet.ActiveCards.FirstOrDefault(c => c == new DynamiteCardType());
```
Comment (IvanovAndrew) [inline Bang/Game/Gameplay.cs:158]: `PlayerTurn.ActiveCards.FirstOrDefault(c => c == new DynamiteCardType())` would be shorter

File: Bang/Game/Gameplay.cs [hunk 4]
```diff
@@ -150,5 +150,61 @@ public void ChooseCard(BangGameCard card, Player player)
         {
             state = state.ApplyReplyAction(player, card);
         }
+
+        public void StartNextPlayerTurn()
```
Comment (IvanovAndrew) [inline Bang/Game/Gameplay.cs:154]: Method `StartNextPlayerTurn()` looks too long. I suppose to inroduce two methods: first one will handle dynamite card logic, second one will handle jail card logic.

File: Bang/GameEvents/CardEffects/JailActionHandler.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        public override HandlerState ApplyEffect(Game.Gameplay gameplay, Player victim, BangGameCard card)
+        {
+            if (victim.Role == new Sheriff())
+            {
+                Logger.Info("Can't play Jail card on Sheriff!");
+                return new ErrorState();
+            }
+
+            Logger.Info($"Player {gameplay.PlayerTurn.Character.Name} player Jail card on {victim.Character.Name}");
```
Comment (IvanovAndrew) [inline Bang/GameEvents/CardEffects/JailActionHandler.cs:18]: Small typo: _player Jail card_. Should be played or plays

File: Bang.Tests/DynamiteSpecification.cs [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+            // act
+            actor.PlayCard(DynamiteCard());
+
+            // Assert
+            actor.PlayerTablet.ActiveCards.Should().Contain(DynamiteCard());
+        }
+
+        [Fact]
+        public void Player_with_dynamite_card_lose_3_life_points_if_dynamite_explodes()
```
Comment (IvanovAndrew) [inline Bang.Tests/DynamiteSpecification.cs:44]: _Player ... lose 3 life points_ looks like small typo

Comment (bashnia007) [reply Bang.Tests/DynamiteSpecification.cs:44]: done

File: Bang/GameEvents/CardEffects/JailActionHandler.cs (hunk 5 above)
Comment (bashnia007) [reply Bang/GameEvents/CardEffects/JailActionHandler.cs:18]: done

File: Bang/Game/Gameplay.cs (hunk 3 above)
Comment (bashnia007) [reply Bang/Game/Gameplay.cs:158]: done

File: Bang/GameEvents/CardEffects/DynamiteActionHandler.cs (hunk 2 above)
Comment (bashnia007) [reply Bang/GameEvents/CardEffects/DynamiteActionHandler.cs:11]: done

File: Bang/Game/Gameplay.cs (hunk 4 above)
Comment (bashnia007) [reply Bang/Game/Gameplay.cs:154]: done

File: Bang/Players/Player.cs (hunk 1 above)
Comment (bashnia007) [reply Bang/Players/Player.cs:150]: done

**Pull request merged**